src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -163,24 +163,12 @@ internal static CSharpType FromSystemType(Type type, string defaultNamespace, So
         internal static CSharpType FromSystemType(BuildContext context, Type type)
             => FromSystemType(type, context.DefaultNamespace, context.SourceInputModel);
 
-        public bool IsCollectionType()
-        {
-            if (!IsFrameworkType)
-                return false;
-
-            return FrameworkType.Equals(typeof(IList<>)) ||
-                FrameworkType.Equals(typeof(IEnumerable<>)) ||
-                FrameworkType == typeof(IReadOnlyList<>) ||
-                FrameworkType.Equals(typeof(IDictionary<,>)) ||
-                FrameworkType == typeof(IReadOnlyDictionary<,>);
-        }
```

Comment:
I remove this method because we already have a static method `IsCollectionType` in `TypeFactory` doing the same thing and it is already widely used and this method only have one usage.